ROADMAP.md

```diff
@@ -0,0 +1,111 @@
+Short-term goals are those which we have committed to and have planned an expected completion date.
+These are organised into milestones.
+Each milestone has,
+
+- A description of it's aims
+- A target date
+- A collection of issues which constitute the work to be done
+
+Milestones will usually correspond to new releases of Data Safe Haven.
+Currently we aim to align milestones with Data Study Groups at the Turing, so that new releases are made in time to be used at these events.
+
+The milestones can be seen [on GitHub](https://github.com/alan-turing-institute/data-safe-haven/milestones)
+
+## Long Term
+
+Long-term goals are developments which we have committed to but have not expected completion date.
+This may be because they are lower priority than short-term goals or because the time required needs to be scoped.
+
+## Desired Features
+
+These are features we would like to highlight but which we have actively decided not to incorporate into the long- or short-term roadmap.
+This may be because, for example, we have decided we have insufficient resource, are lacking the right expertise or it is low priority.
+
+### Develop Outside, Run Inside
+
+A workflow enabling researchers to develop this research and analysis code outside of the TRE, using whatever tools they prefer and a familiar with, and bring this work into the TRE to run against the sensitive data.
+
+This would improve,
+
+- Reproducibility (as users would be encouraged and enabled to use public VCS repositories)
+- User experience
+- User efficiency
+- Accuracy of results (as testing code is easier)
+
+This would involve,
+
+- A trusted mechanism to bring code into the TRE
+- Synthetic or dummy data for testing outside of the TRE
+
+### User Defined Container Support
+
+Enable users to bring containers they have developed outside of the TRE into the TRE to be used in research/analysis.
+The HPC focused container project Apptainer could be a good fit, providing performance and protecting against privilege escalation.
+This work would compliment [Develop Outside, Run Inside](#develop-outside-run-inside).
+
+### Improve Resource Competition
+
+Researchers currently tend to share resources in SRDs which can cause problems if they are not careful to coordinate their work.
+This could be improved by,
+
+- HPC style job submission queue
+- Isolated resources per user (VMs, cgroups)
+
+### User-Facing Web App for Project Management
+
+There is currently little tooling for research teams working through processes or managing their environments.
+A web app could be an accessible way to present features like,
+
+- Switching on and off resources (improves cost management)
+- Scaling resources
+- Opening support tickets
+- Opening data ingress or egress request tickets
+- Working through IG processes
+
+This work would be particularly powerful if the interfaces/processes can be abstracted so that the web app can be used by other TRE operators and integrated into their TRE implementations.
```

Comment:
It could be worth mentioning that. I don't think it is quite at the scale of the goal here though.

Maybe I'm a little hesitant to direct people to an inactive project too.

Reply:
I think it can be helpful to say 'hey here's some work we've done which you could take and repurpose if you want'

Reply:
I don't want to say that precisely because the web app isn't what the section describes and because we aren't maintaining it.

It's a bit of a poisoned chalice to point people towards a project they would have to adopt themselves.

Reply:
Agree with Jim, the standalone IG app is really quite a different thing to what is proposed here. This is something a little more like Azure TRE's UI.

Reply:
Yes, the webapp does at best 1 of these 5 things.

Reply:
Ok, let's leave this for now.